flowkit_ui_backend/impl/data_api_impl.py

```diff
@@ -569,6 +569,21 @@ async def generate_signed_dqs_url() -> SignedUrl:
 
 async def get_latest_date(pool: Pool) -> LatestDate:
     sql = f"SELECT max(dt) FROM `metadata`;"
+    sql = """
```

Comment:
I think I'd bin the with and do something like `select dt from metadata left join scope_mapping USING (mdid) where scope='read:premium_data' ORDER BY dt DESC limit 1`